apps/store/themes/store/js/assets.js

```diff
@@ -72,35 +71,55 @@ store.infiniteScroll.getItems = function(from,to){
     // Returns the jQuery ajax method
     var url = caramel.tenantedUrl(store.asset.paging.url+"&paginationLimit=" + to + "&start="+from+"&count="+count+store.infiniteScroll.setQueryParams(path));
 
-    if(!firstRun) {
+    if(!store.firstRun) {
         caramel.render('loading', 'Loading assets.', function (info, content) {
             $('.loading-animation-big').remove();
             $('body').append($(content));
             $('.loading-animation-big').css('bottom','48px').css('left','0');
         });
     } else {
-        firstRun = false;
+        store.firstRun = false;
     }
-
-        caramel.data({
-             title : null,
-             body : ['assets']
-         }, {
-             url : url,
-             success : function(data, status, xhr) {
-                 if(data.body.assets.context.assets.length == 0) doPagination = false;
-                 caramel.partials(data._.partials, function() {
-                     caramel.render('assets-thumbnails', data.body.assets.context, function (info, content) {
-                         $('.assets-container section').append($(content));
-                         $('.loading-animation-big').remove();
-                     });
-                 });
-             },
-             error : function(xhr, status, error) {
-                 $('.loading-animation-big').remove();
-                 doPagination = false;
-             }
-         });
+    caramel.data({
+        title : null,
+        body : ['assets']
+    }, {
+        url : url,
+        success : function(data, status, xhr) {
+            if(data.body.assets.context.assets.length == 0) store.doPagination = false;
+            caramel.partials(data._.partials, function() {
+                caramel.render('assets-thumbnails', data.body.assets.context, function (info, content) {
+                    $('.assets-container section').append($(content));
+                    if(data.body.assets.context.assets.length != 0){
+                        $('.top-assets-empty-assert').remove();
+                    }
+                    $('.sort-asset-container').show();
+                    $('.loading-animation-big').remove();
+                });
+                if(data.body.assets.context.assets.length == 0
```

Comment:
Fixed in https://github.com/wso2/carbon-store/commit/0424453c4bc5e31408bc123636add41c604a4ef1